src/midi.rs

```diff
@@ -83,54 +95,82 @@ fn get_ports(midi_in: &MidiInput, print: bool) -> Result<std::vec::Vec<midir::Mi
 pub struct MidiConnection {
     port: MidiInputPort,
     port_name: String,
-    midi_in: Option<MidiInput>
+    midi_in: Option<MidiInput>,
+}
+
+fn port_prompt(ports: &Vec<MidiInputPort>, re_ask: Option<()>) -> usize {
+    let mut choice: usize = 0;
+    if let None = re_ask {
+        println!("Please choose your midi input.");
+    }
+    match read_input().expect("Couldn't read input.").parse::<usize>() {
+        Ok(c) => {
+            if c < ports.len() {
+                println!("You have chosen {}", c);
+                choice = c;
+            } else {
+                println!("Please enter a valid number of a port.");
+                port_prompt(ports, Some(()));
+            }
+        }
+        Err(_) => {
+            println!("Please enter a number of the port you want to choose instead.");
+            port_prompt(ports, Some(()));
+        }
+    };
+    choice
 }
 
 impl MidiConnection {
-    pub fn try_new(port: usize) -> Result<MidiConnection, Box<dyn Error>> {
+    pub fn try_new() -> Result<MidiConnection, Box<dyn Error>> {
         let mut midi_in = MidiInput::new("rust-synth input")?;
         midi_in.ignore(Ignore::None);
-        
+
         let ports = get_ports(&midi_in, true)?;
-    
+        let port = port_prompt(&ports, None);
+
         let port = ports.get(port).ok_or("port not found")?.clone();
         let port_name = midi_in.port_name(&port)?;
-    
+
         println!("Selecting: {}", port_name);
```

Comment:
```suggestion
        println!("Selected: {}", port_name);
```
Using the infinitive makes you question is the action is done or still going